dataquality/dq_start/__init__.py

```diff
@@ -0,0 +1,381 @@
+        """
+        pass
+
+
+def detect_model(model: Any, framework: Optional[ModelFramework]) -> Type[BaseInsights]:
+    """Detect the model type in a lazy way and return the appropriate class.
+    :param model: The model to inspect, if a string, it will be assumed to be auto
+    :param framework: The framework to use, if provided it will be used instead of
+         the model
+    """
+    if hasattr(model, "pipe") or framework == ModelFramework.spacy:
+        return SpacyInsights
+    elif hasattr(model, "fit") or framework == ModelFramework.keras:
+        return TFInsights
+    elif hasattr(model, "register_forward_hook") or framework == ModelFramework.torch:
+        return TorchInsights
+    elif hasattr(model, "push_to_hub") or framework == ModelFramework.hf:
+        return TrainerInsights
+    elif isinstance(model, str) or framework == ModelFramework.auto:
+        return AutoInsights
+    else:
+        raise GalileoException(
+            f"Model type: {type(model)} could not be detected model type"
+        )
+
+
+class DataQuality:
+    def __init__(
+        self,
+        model: Optional[Any] = None,
+        task: TaskType = TaskType.text_classification,
+        labels: Optional[List[str]] = None,
+        train_data: Optional[Any] = None,
+        test_data: Optional[Any] = None,
+        val_data: Optional[Any] = None,
+        project: str = "",
+        run: str = "",
+        framework: Optional[ModelFramework] = None,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        """DataQuality
+        :param model: The model to inspect, if a string, it will be assumed to be auto
+        :param task: Task type for example "text_classification"
+        :param project: Project name
+        :param run: Run name
+        :param train_data: Training data
+        :param test_data: Optional test data
+        :param val_data: Optional: validation data
+        :param labels: The labels for the run
+        :param framework: The framework to use, if provided it will be used instead of
+            inferring it from the model. For example, if you have a spacy model, you
+            can pass framework="spacy". If you have a torch model, you can pass
+            framework="torch"
+        :param args: Additional arguments
+        :param kwargs: Additional keyword arguments
+
+        .. code-block:: python
+
+            from dataquality import DataQuality
+
+            with DataQuality(model, "text_classification",
+                             labels = ["neg", "pos"],
+                             train_data = train_data) as dq:
+                model.fit(train_data)
+
+        If you want to train without a model, you can use the auto framework:
+
+        .. code-block:: python
+
+            from dataquality import DataQuality
+
+            with DataQuality(labels = ["neg", "pos"],
+                             train_data = train_data) as dq:
+                dq.finish()
```

Comment:
i still really don't like this syntax for auto. It seems only worse than just `dq.auto()`
```suggestion
            import dataquality as dq

            with dq(train_data = train_data):
                dq.finish()
```